zyte_common_items/pipelines.py

```diff
@@ -53,3 +55,43 @@ def __init__(self):
 
     def process_item(self, item, spider):
         return ae.downgrade(item)
+
+
+class DropItem(Exception):
+    """Drop item from the item pipeline."""
+
+    pass
+
+
+class DropLowProbabilityItemPipeline:
+    """Drop item with a probability less than threshold."""
+
+    def __init__(self, stats):
+        self.stats = stats
+        self.threshold = None
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        o = cls(crawler.stats)
+        return o
+
+    def get_threshold(self, item, spider):
+        return spider.settings.get("ITEM_PROBABILITY_THRESHOLDS", {}).get(
+            item.__class__.__name__, DEFAULT_ITEM_PROBABILITY_THRESHOLD
```

Comment:
`item.__class__.__name__` is not very reliable. Could we use [global_object_name](https://github.com/scrapy/scrapy/blob/48c5a8c98f545e35708a580e724d0b8e1ada5e6e/scrapy/utils/python.py#L309)(`item.__class__`)?